backend/LexBoxApi/GraphQL/RefreshProjectMembershipInterceptor.cs

```diff
@@ -0,0 +1,92 @@
+using HotChocolate.Configuration;
+using HotChocolate.Resolvers;
+using HotChocolate.Types.Descriptors.Definitions;
+using HotChocolate.Utilities;
+using LexBoxApi.Auth;
+using LexCore.Auth;
+using LexCore.Entities;
+
+namespace LexBoxApi.GraphQL;
+
+public class RefreshJwtProjectMembershipMiddleware(FieldDelegate next, ILogger<RefreshJwtProjectMembershipMiddleware> logger)
+{
+    public async Task InvokeAsync(IMiddlewareContext context)
+    {
+        await next(context);
+        var httpContext = context.GetGlobalStateOrDefault<HttpContext>("HttpContext");
+        if (httpContext?.Response.Headers.ContainsKey(LexAuthService.JwtUpdatedHeader) == true)
```

Comment:
@myieye added this, can you review?

Reply:
I realized last night that this was missing, yes 😆👍. I was planning on putting it in `context.ScopedContextData`, but that would be redundant.

Reply:
It's mostly working, but not entirely, because `RefreshUser` is async.
What do you think about this slightly complicated change? Is it worth it?
![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/987efe1b-9465-4bf5-a17e-7fed8a4f3283)

Reply:
so this is to prevent a race condition where the middleware is being executed more than once at a time? I think in that case we should handle that in the middleware and not here. Maybe using scoped context data instead.

Reply:
It would be nice to have a way for any of our code to check if the token has already been updated 🤔
Then we could have potentially redundant (depending on the query/request) code in all sorts of places without risking performance hits.

Just throwing it out there. I'm not sure what that would look like.

Perhaps the `LoggedInContext` could track that.